object/object.go

```diff
@@ -198,24 +199,34 @@ func WriteStrings(out *strings.Builder, list []Object, before, sep, after string
 func (f Function) Type() Type { return FUNC }
 
 // Must be called after the function is fully initialized.
+// Whether a function result should be cached doesn't depend on the Name,
+// so it's not part of the cache key.
 func (f *Function) SetCacheKey() string {
 	out := strings.Builder{}
 	out.WriteString("func")
+	f.CacheKey = f.finishFuncOutput(&out)
+	return f.CacheKey
+}
+
+// Common part of Inspect and SetCacheKey. Outputs the rest of the function.
+func (f *Function) finishFuncOutput(out *strings.Builder) string {
```

Comment:
A bit out of this PR, but maybe you could use godoc link

```suggestion
// Common part of [Inspect] and [SetCacheKey]. Prints the rest of the function.
```

Reply:
I'm unsure about the godoc link in such case

Reply:
godoc doesn't generate doc for private function does it? or maybe there is an option to do so I suppose

Reply:
Good point